services/galley/src/Galley/Aws.hs

```diff
@@ -0,0 +1,173 @@
+    liftAWS aws = view awsEnv >>= \e -> AWS.runAWS e aws
+
+mkEnv :: Logger -> Manager -> JournalOpts -> IO Env
+mkEnv lgr mgr opts = do
+    let g = Logger.clone (Just "aws.galley") lgr
+    e <- configure <$> mkAwsEnv g
+    q <- getQueueUrl e (opts^.queueName)
+    return (Env e g q (opts^.awsRegion))
+  where
+    mkAwsEnv g =  set AWS.envLogger (awsLogger g)
+               .  set AWS.envRegion (opts^.awsRegion)
+              <$> AWS.newEnvWith AWS.Discover Nothing mgr
+
+    awsLogger g l = Logger.log g (mapLevel l) . Logger.msg . toLazyByteString
+
+    mapLevel AWS.Info  = Logger.Info
+    -- Debug output from amazonka can be very useful for tracing requests
+    -- but is very verbose (and multiline which we don't handle well)
+    -- distracting from our own debug logs, so we map amazonka's 'Debug'
+    -- level to our 'Trace' level.
+    mapLevel AWS.Debug = Logger.Trace
+    mapLevel AWS.Trace = Logger.Trace
+    -- n.b. Errors are either returned or thrown. In both cases they will
+    -- already be logged if left unhandled. We don't want errors to be
+    -- logged inside amazonka already, before we even had a chance to handle
+    -- them, which results in distracting noise. For debugging purposes,
+    -- they are still revealed on debug level.
+    mapLevel AWS.Error = Logger.Debug
+
+    configure = set AWS.envRetryCheck retryCheck
+
+    -- TODO: Remove custom retryCheck? Should be fixed since tls 1.3.9?
+    -- account occasional TLS handshake failures.
+    -- See: https://github.com/vincenthz/hs-tls/issues/124
+    -- See: https://github.com/brendanhay/amazonka/issues/269
+    retryCheck _ InvalidUrlException{} = False
+    retryCheck n (HttpExceptionRequest _ ex) = case ex of
+        _ | n >= 3                    -> False
+        NoResponseDataReceived        -> True
+        ConnectionTimeout             -> True
+        ConnectionClosed              -> True
+        ConnectionFailure _           -> True
+        InternalException x           -> case fromException x of
+            Just TLS.HandshakeFailed {} -> True
+            _                           -> False
+        _                             -> False
+
+    getQueueUrl :: AWS.Env -> Text -> IO QueueUrl
+    getQueueUrl e q = do
+        x <- runResourceT . AWST.runAWST e $
+            AWST.trying AWS._Error $
+                AWST.send (SQS.getQueueURL q)
+        either (throwM . GeneralError)
+               (return . QueueUrl . view SQS.gqursQueueURL) x
+
+execute :: MonadIO m => Env -> Amazon a -> m a
+execute e m = liftIO $ runResourceT (runReaderT (unAmazon m) e)
+
+enqueue :: E.TeamEvent -> Amazon ()
+enqueue e = do
+    QueueUrl url <- view eventQueue
+    res <- retrying (limitRetries 1) (const isTimeout) $ const (sendCatch (req url))
+    either (throwM . GeneralError) (const (return ())) res
+  where
+    event = decodeLatin1 $ B64.encode $ encodeMessage e
+    req url = SQS.sendMessage url event & SQS.smMessageGroupId .~ Just "team.events"
```

Comment:
Are you referring to the `MessageGroupId`? This is only relevant when sending to a FIFO queue, consumers don't need to know about it:

> [Receiving Messages
You can't request to receive messages with a specific message group ID.](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues.html)